Sources/AppReceiptValidator/AppReceiptValidator.swift

```diff
@@ -152,10 +152,47 @@ private extension AppReceiptValidator {
     func checkSignatureAuthenticity(pkcs7: ASN1Decoder.PKCS7, appleRootCertificateData: Data, rawData: Data?) throws {
         guard let signature = pkcs7.signatures?.first else { throw Error.receiptNotSigned }
         guard let signatureData = signature.signatureData else { throw Error.receiptNotSigned }
+        guard let appStoreCertFromReceipt = pkcs7.certificates.first else { throw Error.receiptMissingCertificates }
         guard let receiptData = pkcs7.mainBlock.findOid(.pkcs7data)?.parent?.sub?.last?.sub(0)?.rawValue else { throw Error.receiptNotSigned }
 
-        let rootCert = pkcs7.certificates[0]
-        try self.verifyAuthenticity(x509Certificate: rootCert, receiptData: receiptData, signatureData: signatureData)
+        try self.verifyCertificates(pkcs7: pkcs7, appleRootCertificateData: appleRootCertificateData)
+        try self.verifyAuthenticity(x509Certificate: appStoreCertFromReceipt, receiptData: receiptData, signatureData: signatureData)
+    }
+
+    func verifyCertificates(pkcs7: ASN1Decoder.PKCS7, appleRootCertificateData: Data) throws {
+        let appleRootCertificate = try X509Certificate(data: appleRootCertificateData)
+        guard let rootCertFromReceipt = pkcs7.certificates.last else { throw Error.receiptMissingCertificates }
+        guard let expectedRootCertKey = appleRootCertificate.publicKey?.derEncodedKey else { throw Error.appleRootCertificateNotFound }
+        guard let rootCertFromReceiptKey = rootCertFromReceipt.publicKey?.derEncodedKey else { throw Error.receiptMissingCertificates }
+
+        if rootCertFromReceiptKey != expectedRootCertKey {
```

Comment:
We compare the root certificate's key we found in the receipt with the one we are expecting. 
This can also work on linux.

Reply:
Nice additions - Didn't realize that we got direct access to the `derEncodedKey` by adopting ASN1Decoder. 
Maybe we should note in the README that we skip the key comparison on Linux.

Reply:
Yes, this was available for the public key, but for the certificates themselves I added an accessor to the `derData` to our fork of `ASN1Decoder`